frontend/src/components/Auth/Login.tsx

```diff
@@ -2,38 +2,43 @@ import React, { useEffect, useState } from 'react'
 import './login.css'
 import { LoginAPI } from '../../APIs/Auth/Login';
 import { useNavigate } from 'react-router-dom';
+import Message from './Message';
 function Login() {
     const [username, setUsername] = useState('')
     const [password, setPassword] = useState('')
     const [errors, setErrors] = useState({
         username: '',
         password: ''
     });
-    const navigate = useNavigate();
     const [success, setSuccess] = useState(false);
+    const [hasAttemptedLogin, setHasAttemptedLogin] = useState(false);
```

Comment:
we may not need this extra state, we can change the success state into status
```suggestion
type Status = "success" | "error" | "idle"
    const [status, setStatus] = useState<Status>("idle"
```